src/IECoreDelight/Renderer.cpp

```diff
@@ -1104,25 +1135,28 @@ class DelightRenderer final : public IECoreScenePreview::Renderer
 					m_oversampling = 9;
 				}
 			}
+			else if(
+				name == g_maxLengthDiffuseOptionName ||
+				name == g_maxLengthHairOptionName ||
+				name == g_maxLengthReflectionOptionName ||
+				name == g_maxLengthRefractionOptionName ||
+				name == g_maxLengthSpecularOptionName ||
+				name == g_maxLengthVolumeOptionName ||
+				name == g_clampIndirectOptionName
+			)
+			{
+				// These parameters fail to set properly if they are not doubles
+				auto data = runTimeCast<const FloatData>( value );
```

Comment:
It seems a pity to ignore DoubleData if it is provided by the caller already - in fact we're now deletiing the option if DoubleData is provided.

Reply:
Fixed in refactor in https://github.com/GafferHQ/gaffer/pull/5333/commits/0ca905e98abe63931686805e3f9a5d1ae018d175